### ISSUE TITLE
POC: Implement EC2 Instance Connect Endpoint (EIC Endpoint) locally

### ISSUE THREAD

**Issue by yaseenzafar123** [XgridInc/xc3#37]
**User stories**

As a cloud engineer, I want to implement EC2 Instance Connect Endpoint (EIC Endpoint) in our internal AWS account so that private instances can be accessed without bastion hosts.

**Acceptance Criteria**

- [x] Ensure that you have the necessary IAM permissions
- [x] Identify the private EC2 instances that you want to implement EIC Endpoiint
- [x] Do the necessary configurations and test the system
- [x] Update the steps in documentation form

**Functional requirements**

- [x] Use AWS CLI to configure EC2 Instance Connect Endpoint for the AWS account.
- [x] Ensure that SSH access to EC2 instances is restricted to IAM users with the necessary permissions.
- [x] Validate that EC2 instances can be accessed securely using EC2 Instance Connect through CLI

**Comment by yaseenzafar123** [comment]
19 June 2023, Monday:

Did R&D on how to implement EIC Endpoint. Went through the documentation: https://aws.amazon.com/blogs/compute/secure-connectivity-from-public-to-private-introducing-ec2-instance-connect-endpoint-june-13-2023/ and few other videos.

20 June 2023, Tuesday:

Updated AWS CLI to latest version and configured EIC Endpoint Service. Later in the day, I was able to access the private instance using CLI
